src/python/pants/source/source_root.py

```diff
@@ -373,6 +375,6 @@ def find(self, path):
           node = child
           j += 1
       if node.is_terminal:
-        return self._source_root_factory.create(os.path.join(*keys[1:j]), langs, node.category)
+        return self._source_root_factory.create(os.path.join(*keys[1:j]) if j > 1 else '', langs, node.category)
```

Comment:
Would prefer this be broken out as a separate line with a comment, as the ternary isn't very self explanatory.